src/createPrefetch.mts

```diff
@@ -54,6 +66,13 @@ function createPrefetchHook({
                 )
               ),
               ...requestParams,
+              ts.factory.createParameterDeclaration(
```

Comment:
Why have the queryKey parameter?
I believe the queryKey function with passed-in parameters should be enough for the prefetch.

Reply:
@7nohe any comments on this?

Reply:
I believe that having a queryKey is necessary for flexibility. Since normal queries accept a queryKey, we should conform to that as well.

Reply:
I can't come up with a specific case, but if I were to explain using the Next.js example I just added, using keys such as QueryA or QueryB rather than matching tags or limits makes it easier to manage the code.

app/page.tsx
```tsx
import { prefetchUseDefaultServiceFindPets } from "@/openapi/queries/prefetch";
import {
  HydrationBoundary,
  QueryClient,
  dehydrate,
} from "@tanstack/react-query";
import Pets from "./pets";

export default async function Home() {
  const queryClient = new QueryClient();

  await prefetchUseDefaultServiceFindPets(queryClient, {
    limit: 10,
    tags: [],
  }, ['QueryA']);

  await prefetchUseDefaultServiceFindPets(queryClient, {
    limit: 100,
    tags: ['tag1', 'tag2', 'tag3'],
  }, ['QueryB']);

  return (
    <main className="flex min-h-screen flex-col items-center justify-between p-24">
      <HydrationBoundary state={dehydrate(queryClient)}>
        <Pets />
      </HydrationBoundary>
    </main>
  );
}
```

Reply:
Thanks for the example.

I am failing to understand how it is 'easier' than using the generated key and properties.

When overriding, there will be queries and keys that do not match when these hooks are used in other places in code.

I feel it would be safer for the consumer to write their own hooks and preferences so they have complete control rather than us allowing them to configure keys that they would have to override every single use.

Reply:
> Thanks for the example.
> 
> I am failing to understand how it is 'easier' than using the generated key and properties.
> 
> When overriding, there will be queries and keys that do not match when these hooks are used in other places in code.
> 
> I feel it would be safer for the consumer to write their own hooks and preferences so they have complete control rather than us allowing them to configure keys that they would have to override every single use.

As you mentioned, that does seem to be the safer option.  Since this library outputs pure TypeScript clients, it would be better to use that to build your own hooks.
It seems better to allow a queryKey option after finding some practical cases.

Reply:
@nopitown 
Would you remove the `queryKey` parameter?
The latest main branch includes the Next.js sample app. You can run it with the following command. 

```bash
npm run build && pnpm --filter nextjs-app generate:api && pnpm --filter nextjs-app dev   
```

If you're busy, I can make the changes instead.